src/core/Akka/Actor/ActorBase.cs

```diff
@@ -57,11 +57,13 @@ public Failure(Exception cause)
                 Cause = cause;
             }
 
+            /// <summary>
+            /// Checks if a provided failure message was a result of cancelled operation.
+            /// </summary>
+            public bool IsCancelled => Cause == null;
```

Comment:
I approve of these API changes, although we'll need to start telling people in the docs to look for `Status.Failure` messages if they're not handling task failures / cancellations themselves inside their own continuations at the end of a `Task`

Reply:
Although, this check for determining if the task was cancelled... `Cause == null`.... IMHO, this value shouldn't be inferred by a loose convention like this. Rather, we should add a constructor overload to `Status.Failed` where we explicitly set this value to true or false and mark the old constructor as `Obsolete`. 

Right now we're assuming that having a null exception means a task cancellation - might be right most of the time but that's not good enough. Needs to be intentional. We should change the API to add an explicit overload and handle that inside the `PipeTo` implementation itself.

Reply:
I think, that we could remove that field and simply let users know (through docs) that in case of cancellation exception should be null (+ keep `TaskCancellationException` filler in Ask<>). Setting up separate cancellation handler in PipeTo is bad - it breaks backwards compatibility (currently a null exception is already passed to PipeTo's failure handler on cancelled task).